src/analysis/sum_stat.py

```diff
@@ -0,0 +1,205 @@
+
+        return previous_ss
+
+    def __normalise_pvalues(self, df: pandas.DataFrame, ss_info: dict):
+        for pvalue in ss_info['p']:
+            df[pvalue] = (df[pvalue] - df[pvalue].min()) / \
+            (df[pvalue].max() - df[pvalue].min())
+
+    def prepare_data(self, *ss_and_tag: tuple[str], working_dir: str, to_file='ss.tsv', norm=True) -> pandas.DataFrame:
+        '''
+        Returns aggregated and intersected summary statistics dataframe.
+        All summary statistics files must contain position column and at least one of them should have id column.
+        All necessary columns will be detected automatically with regex.
+
+                Parameters:
+                        ss_and_tag (tuple): Any number of tuples with format: (path_to_summary_stat.tsv, ancestry_tag)
+                        working_dir (str): Path to a dir to store all lists into
+                        to_file (str or None): Path to output dataframe, if None won't create one
+                        norm (bool): Perform min max normalisation for each p-value column separately
+
+                Returns:
+                        merged_df (pandas.DataFrame): Dataframe with position, id and all -log10(p-values) columns from all input files
+        '''
+        ss_info = {}
+        for i, ss_tag in enumerate(ss_and_tag):
+            ss, tag = ss_tag
+            with open(ss, 'r') as ss_file:
+                header = ss_file.readline().strip('\n').split('\t')
+
+            logging.info(f'Reading basic info about {ss}...')
+            ss_info[ss] = self.__get_ss_info(header, tag)
+
+            logging.info(f'Preparing list of positions for {ss}...')
+            pos_list_dist = os.path.join(working_dir, f'pos{i}.list')
+            subprocess.call("awk -F '\t' '{ print $%s }' %s | sort | uniq > %s"
+                            % (ss_info[ss]['pos_i'], ss, pos_list_dist),
+                            shell=True)
+
+        logging.info(f'Making lists of positions intersection...')
+        intersection_list_dist = os.path.join(working_dir, f'intersection.list')
+        subprocess.call("cat pos* | sort | uniq -c | awk '{if($1==%s){print $2}}' > %s"
+                        % (len(ss_info), intersection_list_dist),
+                        shell=True)
+
+        df_dict = {}
+        found_ids = False
+        no_ids_list = []
+        for i, ss_tag in enumerate(ss_and_tag):
+            ss, tag = ss_tag
+            logging.info(f'Filtering rows based on intersection list for {ss}...')
+            subprocess.call("awk 'NR==FNR{a[$1]; next} FNR==1 || $%s in a' %s %s > %s"
+                            % (ss_info[ss]['pos_i'],intersection_list_dist, ss, ss + '.filtered'),
+                            shell=True)
+
+            logging.info(f'Creating pandas dataframe for {ss}...')
+            df = self.__get_dataframe(ss, ss_info)
+
+            if norm:
+                logging.info(f'Performing min max normalisation for {ss}...')
```

Comment:
the problem with this normalisation is that if one of the summary statistics is in -log10 scale and one isn't, you won't find  that out after normalising